### PR TITLE
[MRG] more documentation updates

### PR DESCRIPTION
**Pull request by JuliaSprenger**
[doc] add comment about lazy being deprecated

### CODE REVIEW

**Comment by samuelgarcia** [comment]
Yeah. Promoting next feature!!

Maybe we could also put the information in the changelog. **lazy** option is depreciated in actual behavior (lazy_shape)

**Comment by samuelgarcia** [comment]
OK for me.

Who is the bad guy who had left so much pep8 error in the file  neo/io/basefromrawio.py ?

**Comment by JuliaSprenger** [comment]
Ok, I think that's all the changes I have right now. Any comments or suggestions? Otherwise you can merge.

File: doc/source/io.rst [hunk 1]
```diff
@@ -119,6 +119,8 @@ but all the metadata will be loaded. The *lazy_shape* attribute is added to all
 In this case, *lazy_shape* is a tuple that has the same value as *shape* with ``lazy=False``.
 To know if a class supports lazy mode use ``ClassIO.support_lazy``.
 By default (if not specified), ``lazy=False``, i.e. all data is loaded.
+The lazy option will be removed in future neo versions. Similar functionality will be
```
Comment (apdavison) [inline doc/source/io.rst]: should be "Neo"

File: doc/source/io.rst [hunk 2]
```diff
@@ -119,6 +119,8 @@ but all the metadata will be loaded. The *lazy_shape* attribute is added to all
...
+implemented using ProxyObjects.
```
Comment (apdavison) [inline doc/source/io.rst]: since there is no `ProxyObject` class in Neo 0.6, we should say "proxy objects".

**Pull request merged**